src/entities/course.entity.ts

```diff
@@ -0,0 +1,40 @@
+import { Difficulty } from '../common/enums/difficulty.enum';
+import { SharedEntity } from '../common/model/sharedEntity';
+import { Column, Entity } from 'typeorm';
+
+@Entity()
+export class Course extends SharedEntity {
+  @Column()
+  title: string;
+
+  @Column()
+  description: string;
+
+  @Column()
+  price: number;
+
+  @Column({ nullable: true, default: 0 })
+  rating: number;
+
+  @Column()
+  tutor: string;
+
+  @Column({
+    type: 'enum',
+    enum: Difficulty,
+    default: Difficulty.Easy,
+  })
+  difficulty: Difficulty;
+
+  @Column()
+  duration: number;
+
+  @Column('text', { array: true, nullable: true, default: [] })
+  topics: string[];
+
+  @Column('text', { array: true, nullable: true, default: [] })
+  students: string[];
+
```

Comment:
i think the only compulsory fields should be title, description, and price

Reply:
what do you think?

Reply:
That means it will be fine to have a course without the tutor's name? Language can be nullable